e2e_playwright/websocket_disconnect_test.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from playwright.sync_api import Page, expect
+
+
+def test_disconnecting_disables_widgets_correctly(app: Page):
+    expect(app.get_by_test_id("stButton").locator("button")).not_to_have_attribute(
+        "disabled", ""
+    )
+    expect(app.get_by_test_id("stMarkdown").first).to_contain_text("Value 1: 25")
+
+    expect(app.get_by_test_id("stConnectionStatus")).not_to_be_visible()
+    app.evaluate("window.streamlitDebug.shutdownRuntime()")
```

Comment:
nit: maybe add a comment here that calling this will disconnect the WebSocket connection -> which should put all elements into `disabled` mode.